src/sentry/snuba/events.py

```diff
@@ -232,18 +233,45 @@ class Columns(Enum):
     )
 
 
+class BaseColumn(object):
+    def get(self):
+        raise NotImplementedError
+
+
+class BuiltInColumn(BaseColumn):
+    def __init__(self, col):
+        self.__col = col
+
+    def get(self, dataset=Dataset.Discover):
+        if dataset == Dataset.Events:
+            return self.__col.value.event_name
+        if dataset == Dataset.Transactions:
+            return self.__col.value.transaction_name
+        return self.__col.value.discover_name
+
+
+class TagColumn(BaseColumn):
+    def __init__(self, key):
+        self.__key = key
+
+    def get(self, dataset=Dataset.Discover):
+        return "tags[{}]".format(self.__key)
+
+
+column_map = {col.value.alias: col for col in Columns}
+
+
+def get_column_from_alias(alias):
```

Comment:
Does this need a dataset parameter or is it intended only for the discover dataset?

Reply:
It doesn't because it returns the column object that is not yet resolved to a dataset. However I think it now makes more sense to close this PR in favour of https://github.com/getsentry/sentry/pull/16021 as we can actually do away with this function completely.